### ISSUE TITLE
🐛 [Firebase_messaging] onMessage is called twice on android

### ISSUE THREAD

**Issue by bilalgodesto** [firebase/flutterfire#8841]
## Bug report

**Describe the bug**
I have added firebase messaging and it is working fine but on android onMessage is called twice.
I have tried to use workaround but it is not possible to find any work around for this.
When ever i receive a message. onMessage.listen is called twice.
``` FirebaseMessaging.onMessage.listen((RemoteMessage event) async { ```

### Steps to reproduce
Create a project ` flutter create . `
Add firebase_messaging: 11.4.1 and firebase_core 1.17.1 
Add code fore listening to notification when app is in foreground.

```
FirebaseMessaging.onMessage.listen((RemoteMessage event) async { 
print('received notification');
});

```

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 2.10.3, on macOS 12.1 21C52 darwin-arm, locale
    en-GB)
    • Flutter version 2.10.3 at /Users/bilalrabbi/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (3 months ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/bilalrabbi/Library/Android/sdk
    • Platform android-32, build-tools 30.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.42.0

[✓] Connected device (2 available)
    • iPhone 8 (mobile) • 3AA1D2F9-3DD6-45B3-A823-91A56C2B25D6 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • Chrome (web)      • chrome                               • web-javascript
      • Google Chrome 102.0.5005.61

[✓] HTTP Host Availability
    • All required HTTP hosts are available
```

</details>

---

**Comment by srihariash999** [comment]
@bilalgodesto  I have had the same issue previously and there are a few duplicates of this exact issue.

In all of them, issue was in the implementation of .onMessage.listen 

Please make sure this method is being called or getting initialized only once in your app. If it is being registered multiple times, onMessage callback will be fired multiple times.

See #6588  #7243

**Comment by darshankawar** [comment]
Please follow-up per above and in attached links for further updates.
Closing as duplicate.
